jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
Indeed, according to the current SSB discussions, each module (SP, SD, CM, and BC) will be a class of its own. Furthermore, there will (most likely) be a class called something like SimPhoNyApplication which will store and maintain the SP, SD, CM, and BC objects. Then only a SimPhoNyApplication object would be provided for the engine. So a lot of changes probably in the near future ...

This would mean, among other things, that the SDs are not directly maintained by the engines themselves. A question related to the current discussion then is, how the "single lattice dataset" can be enforced with jyu-lb engines? Or, in the case of many lattice datasets stored in a SD, which of them is utilized in the simulation?